### PR TITLE
Add CRUD pages for family resource

### LINKED ISSUE #312: As a user, I can view a list of all families in my organization
Acceptance Criteria

- [ ] A new route is added that maps to `Families#index`
- [ ] A new view is added that lists all `Family` records. This list should be scoped to the organization that the `current_user` `belongs_to`
- [ ] A link to this page is added to the navigation bar
- [ ] A basic "smoke test" is added that seeds that database with a few family records (probably 2), loads the page and asserts that these records exist on the page.

There is some prior art for how we want index pages in our application to look. See `Facilities#index` and `Animals#index`. Some of the links that will exist on this page, (new, show, edit, delete). Its fine to leave this blanks and fill those links in later as a separate task, or as a part of the task to add those actions.

### PR DESCRIPTION
**Pull request by todtb**
#

Resolves #312, #314, #316, #318, #320    

### Description
Adds CRUD pages for `Family` resource. 

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

A few feature specs included. New resource routes for `Family` can be navigated to in order to verify.

### CODE REVIEW

File: app/views/families/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
+<section class="section facilities__index">
+  <div class="container facilities__index__container">
+    <p id="notice"><%= notice %></p>
```
Comment (benreyn) [inline app/views/families/index.html.erb:3]: I noted this in #349 but we should make a follow up task to cleanup our usage of `notice` and `altert` and convert to `flashes` as a hash. (no action needed here)

File: app/views/families/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <th colspan="3"></th>
+      </tr>
+      </thead>
+
+      <tbody>
+      <% @families.each do |family| %>
+        <tr>
+          <td><%= family.name %></td>
+          <td><%= link_to family.female.pii_tag, animal_path(family.female) %></td>
+          <td><%= link_to family.male.pii_tag, animal_path(family.male) %></td>
```
Comment (benreyn) [inline app/views/families/index.html.erb:28]: Thoughts on adding delagation to `Family` that will give us `male_pii_tag` and `female_pii_tag` methods on family?

**Review by benreyn — APPROVED** [review]
Looks good! Way to close 5 issues in one PR @todtb !

File: app/views/families/index.html.erb (hunk 2 above)
Comment (todtb) [reply app/views/families/index.html.erb:28]: Sounds good @benreyn. This is going to be the primary identifier for end user use, right? What would the `tag_id` be?

**Comment by todtb** [comment]
Hey @benreyn, changes made, just pending one last co-author confirmation of user/email which I'll amend before we continue.

File: app/views/families/index.html.erb (hunk 2 above)
Comment (benreyn) [reply app/views/families/index.html.erb]: Im working on teasing that out from the data we have from our stakeholders, and digging through legacy code from when I joined the project. Ill follow up on this ASAP.

**Pull request merged**